### DISCUSSION TITLE
(How) can I republish a GitHub Pages site after unpublishing it?

### DISCUSSION THREAD

**Discussion by GrahamHannington** [community/community#142288] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

A week or so ago I published this GitHub Pages site:

https://grahamhannington.github.io/bigas/?text=Your/text/here

using the root folder of the main branch of this personal repo:

https://github.com/GrahamHannington/bigas

That worked.

Earlier today, I pushed a commit to the main branch. I can see the code changes in GitHub.

However, the GitHub Pages site didn't refresh after several minutes.

After clearing my browser cache, and also attempting to manually run the pages build and deployment action (it queued, but has not run), I grew impatient, and decided to try unpublishing the site in the repo settings and then republishing.

That... might have been a mistake.

My repo's Settings > GitHub Pages page reports that:

> Your GitHub Pages site is currently being built from the main branch

but, it's not: attempting to visit the site now shows an error 404.

I can't click "Save" on the option to (re)publish the site: it's grayed-out.

How do I resurrect my GitHub Pages site?

**Comment by GrahamHannington** [comment, marked as answer]
It's up again. I followed [this advice](https://github.com/orgs/community/discussions/28190#discussioncomment-3321541):

> I fix it...try to select branch none, save then enter in the repository- setting-pages and select branch master, save and that´s it, the link of the live site appears again....try it!

I wish I knew why the site wasn't refreshing.

**Reply by zitongzhoueric** [reply]
This worked for me. Had this problem after changing visibility of my pages site to private and then back to public.
